docs/content/guides/dagster/branch-depl-with-snowflake.mdx

```diff
@@ -0,0 +1,369 @@
+
+To ensure we can't accidentally write to production from within our branch deployment, we’ll use a different set of credentials from production and write to our database clone.
+
+Dagster automatically sets certain environment variables containing deployment metadata, allowing us to read these environment variables to discern between deployments. We can access the `DAGSTER_CLOUD_IS_BRANCH_DEPLOYMENT` environment variable to determine the currently executing environment. Refer to [TODO: PAGE TITLE]() for a full list of preset environment variables.
+
+Because we want to configure our assets to write to Snowflake using a different set of credentials and database in each environment, we’ll configure a separate I/O manager for each environment:
+
+```python file=/guides/dagster/local_to_production/branch_depl_tutorial/repository_v1.py startafter=start_repository endbefore=end_repository
+# repository.py
+from dagster import repository, with_resources
+from dagster_snowflake import build_snowflake_io_manager
+from dagster_snowflake_pandas import SnowflakePandasTypeHandler
+import os
+
+snowflake_io_manager = build_snowflake_io_manager([SnowflakePandasTypeHandler()])
+
+
+@repository
+def repo():
+    snowflake_config = {
+        "account": "abc1234.us-east-1",
+        "user": "system@company.com",
+        "password": {"env": "SYSTEM_SNOWFLAKE_PASSWORD"},
+        "schema": "HACKER_NEWS",
+    }
+    resource_defs = {
+        "branch": {
+            "snowflake_io_manager": snowflake_io_manager.configured(
+                {
+                    **snowflake_config,
+                    "database": f"PRODUCTION_CLONE_{os.getenv('DAGSTER_CLOUD_PULL_REQUEST_ID')}",
+                }
+            ),
+        },
+        "production": {
+            "snowflake_io_manager": snowflake_io_manager.configured(
+                {
+                    **snowflake_config,
+                    "database": "PRODUCTION",
+                }
+            ),
+        },
+    }
+
+    def get_current_env():
+        is_branch_depl = os.getenv("DAGSTER_CLOUD_IS_BRANCH_DEPLOYMENT")
+        assert is_branch_depl != None  # env var must be set
+        return "branch" if is_branch_depl else "prod"
+
+    return [
+        with_resources([items, comments, stories], resource_defs=resource_defs[get_current_env()]),
+    ]
+```
+
+---
+
+## Step 3: Clone a production database per branch deployment
+
+The `branch_deployments.yml` workflow file from the [quickstart template](https://github.com/dagster-io/dagster-cloud-branch-deployments-quickstart) defines a `dagster_cloud_build_push` job that builds each branch deployment. In order to add this workflow file, it must be located in your project directory under `.github/workflows`. We can modify this workflow file to add a custom step that launches a job within our repository that will clone the production database upon deployment.
```

Comment:
there's an implication here that they need to get the branch_deployments.yaml from the quickstart template, but if they are on serverless or followed the setup guide then they already have the file 

```suggestion
The `branch_deployments.yml` workflow file in `.github/workflows` defines a `dagster_cloud_build_push` job that builds each branch deployment. We can modify this workflow file to add a custom step that launches a job within our repository that will clone the production database upon deployment.
```